base/sparse/umfpack.jl

```diff
@@ -391,26 +389,29 @@ function nnz(lu::UmfpackLU)
 end
 
 ### Solve with Factorization
-A_ldiv_B!{T<:UMFVTypes}(lu::UmfpackLU{T}, b::Vector{T}) = solve(lu, b, UMFPACK_A)
-A_ldiv_B!{T<:UMFVTypes}(lu::UmfpackLU{T}, b::Matrix{T}) = solve(lu, b, UMFPACK_A)
-function A_ldiv_B!{Tb<:Complex}(lu::UmfpackLU{Float64}, b::Vector{Tb})
-    r = solve(lu, [convert(Float64,real(be)) for be in b], UMFPACK_A)
-    i = solve(lu, [convert(Float64,imag(be)) for be in b], UMFPACK_A)
-    Tb[r[k]+im*i[k] for k = 1:length(r)]
-end
-
-Ac_ldiv_B!{T<:UMFVTypes}(lu::UmfpackLU{T}, b::VecOrMat{T}) = solve(lu, b, UMFPACK_At)
-function Ac_ldiv_B!{Tb<:Complex}(lu::UmfpackLU{Float64}, b::Vector{Tb})
-    r = solve(lu, [convert(Float64,real(be)) for be in b], UMFPACK_At)
-    i = solve(lu, [convert(Float64,imag(be)) for be in b], UMFPACK_At)
-    Tb[r[k]+im*i[k] for k = 1:length(r)]
-end
-
-At_ldiv_B!{T<:UMFVTypes}(lu::UmfpackLU{T}, b::VecOrMat{T}) = solve(lu, b, UMFPACK_Aat)
-function At_ldiv_B!{Tb<:Complex}(lu::UmfpackLU{Float64}, b::Vector{Tb})
-    r = solve(lu, [convert(Float64,real(be)) for be in b], UMFPACK_Aat)
-    i = solve(lu, [convert(Float64,imag(be)) for be in b], UMFPACK_Aat)
-    Tb[r[k]+im*i[k] for k = 1:length(r)]
+for (f!, umfpack) in ((:A_ldiv_B!, :UMFPACK_A),
+                      (:Ac_ldiv_B!, :UMFPACK_At),
+                      (:At_ldiv_B!, :UMFPACK_Aat))
+    @eval begin
+        $f!{T<:UMFVTypes}(x::VecOrMat{T}, lu::UmfpackLU{T}, b::VecOrMat{T}) = solve!(x, lu, b, $umfpack)
+        $f!{T<:UMFVTypes}(lu::UmfpackLU{T}, b::Vector{T}) = $f!(similar(b), lu, b)
+        $f!{T<:UMFVTypes}(lu::UmfpackLU{T}, b::Matrix{T}) = $f!(similar(b), lu, b)
+
+        function $f!{Tb<:Complex}(x::Vector{Tb}, lu::UmfpackLU{Float64}, b::Vector{Tb})
+            n = size(b, 1)
+            # TODO: Optionally let user allocate these and pass in somehow
```

Comment:
I like that idea, especially that it makes the syntax uniform across functions that need different numbers temporary arrays.